fix(networking): prevent double /p2p/ in our addr

b-zee opened the pull request:
Sometimes when we start listening via a relay server, we would start
adding our own address that included a double `/p2p/<peer ID>` to our
external addresses.

dirvine reviewed (approved):
Yea nice one. I like the 2 versions of insert as opposed to some trickery.

oetyng commented:
bors r+

b-zee committed:
fix(networking): prevent double /p2p/ in our addr

Sometimes when we start listening via a relay server, we would start
adding our own address that included a double `/p2p/<peer ID>` to our
external addresses.
